docs/design/coding-conventions.md

```diff
@@ -314,13 +314,58 @@ Note that this also applies to uses of "ID" as an abbreviation for "identifier"
 
 Prefixes based on types (e.g., `p` for pointers) should never be used.
 
-Global variables should have a `g` prefix, e.g. `gCounter`.
-Non-`const` `static` class members can have an `s` prefix if that suits your fancy.
+Global variables should have a `g_` prefix, e.g. `g_counter`.
+Non-`const` `static` class members can have an `s_` prefix if that suits your fancy.
 Of course, both of these should be avoided, so this shouldn't come up often.
 
 Constant data (in the sense of `static const`) should have a `k` prefix.
 
-In contexts where "information hiding" is relevant/important, such as when a type has both `public` and `private` members, or just has certain operations/fields that are considered "implementation details" that most clients should not be using, and `_` prefix on function and field members is allowed (but not required).
+In contexts where "information hiding" is relevant/important, such as when a type has both `public` and `private` methods, or just has certain operations that are considered "implementation details" that most clients should not be using, using `_` prefix on lower camel functions is allowed (but not required).
+
+Member variables can be prefixed with `m_` to identify them as such. As a rule of thumb, if a class/struct has member functions and/or constructors then prefixing should be used. The major benefit being the clarity in code of member access over parameters or local variables. This also avoids the clash of parameter names and member variables that typically occurs otherwise in constructors. For example...
+
+```c++
+// In this example it has methods and value which are 
+class SomeClass 
+{
+    public:
+    int getValue() const { return m_value; }
+    void incrementValue() { m_value += s_increment; }
+        /// Public interface method calling an internal implementation 
+    bool isOdd() const { return _isOdd(); }
+    
+    static void setIncrement(int inc) { s_increment = inc; }
+    
+        /// There is no confusion over the parameter and the member variable
+    explicit SomeClass(int value = 0):
+        m_value(value)
+    {}
+    
+    private:
+    // Because this method is 'internal implementation' it is prefixed with _
+    bool _isOdd(int change) const { return (m_value & 1) != 0; }
+    
+    // This member value is prefixed with m_ because class has protection and methods
+    int m_value;
+    // Static members can be prefixed with s_
+    static int s_increment;
+};
+
+/* static */int SomeClass::s_increment = 1;
+```
+
+For classes that don't have methods the class is typically acting as a 'bag of values' that are directly accessed, and so the arguments for prefixing don't apply. For example...
+
+```c++
+// Typically 'bag of values' should be structs - because they have no protection
+struct AnotherClass
+{
+    // The members are not prefixed, as there are no methods/ctors/dtors. 
```

Comment:
This example is good to have to make the policy for "bag of values" types clear, but I don't agree with the rationale here, since I'd claim that it is often useful to have a "bag of values" type that *does* have methods, constructors, and maybe even a destructor.

For me the choice here has a *lot* less to do with what other features a type has, and more with the definition of the conceptual "property." If it has non-trivial invariants or access rules, then hiding it and defining getter/setter operations makes sense. If there is nothing interesting to protect/preserve, then exposing a plain old field is fine (and saves everybody time/effort). It is natural for some fields to evolve over time and need to be protected later on (because a type gains new invariants or non-trivial encoding tricks), but we can pay the cost of encapsulating them when that happens, rather than up front.

Reply:
That's fine it's a 'rule of thumb', and so if the prefix isn't wanted then so be it. But that is the argument for it - and it is a discussion that we have had in the past (on github no less) that came up with this distinction that I believe was settled. It is not a new idea that I'm just bringing up.

Reply:
I think we've been aligned on the general idea of having plain old member variables in some cases, and decorated ones in others, and this is more about the nuance of when to prefer one convention over the other. I just wouldn't go so far as to say that only truly trivial types should have public non-prefixed members, and if I seemed to agree with that as a cutoff point I must have misunderstood.

Anyway, the line on where to favor one over the other is really down to programmer preference at the end of the day. If somebody decides all their member variables need to be `private` and they want to write their accessors up front, I wouldn't reject that code since once they've written that code the cost is already sunk.